packages/protocol/lib/compatibility/report.ts

```diff
@@ -122,12 +122,22 @@ export interface ASTVersionedReportIndex {
   libraries: CategorizedChangesIndex
 }
 
-export const isLibrary = (contract: string, artifacts: BuildArtifacts) => {
-  const artifact = artifacts.getArtifactByName(contract)
-  const zContract = makeZContract(artifact)
-  const ast = new ContractAST(zContract, artifacts)
-  const kind = ast.getContractNode().contractKind
-  return kind === 'library'
+export const isLibrary = (contract: string, artifactsSet: BuildArtifacts[]) => {
+  for (const artifacts of artifactsSet){
+    // need to get a try here
+
+    const artifact = artifacts.getArtifactByName(contract)
+    if (artifact === undefined){
+      // EAFP
+      // thse artifacts doesn't have the desired property
+      // A priori we don't know in what folder they are
```

Comment:
nit. if this is going to be a permanent comment, then it'll need rewording to enable easier understanding.

Reply:
I am not sure this if will actually be needed after all changes in this PR